src/partr.c

```diff
@@ -538,6 +544,14 @@ JL_DLLEXPORT jl_task_t *jl_task_get_next(jl_value_t *trypoptask, jl_value_t *q,
     }
 }
 
+JL_DLLEXPORT jl_task_t *jl_task_get_next(jl_value_t *trypoptask, jl_value_t *q, jl_value_t *checkempty)
+{
+    uint64_t t0 = jl_record_time_for_tls_metric();
+    jl_task_t *task = _jl_task_get_next(trypoptask, q, checkempty);
+    jl_increment_timing_tls_metric(jl_current_task->ptls, scheduler_time, jl_record_time_for_tls_metric() - t0);
+    return task;
+}
```

Comment:
I think this got lost in the handoff, but the original plan was to _not_ record the current time twice for the "fast-path" of task switching.

The concern is that adding two syscalls can meaningfully slow down task switching, and the idea was that we probably don't need to do this anyway since the time we'd be measuring in the fast-path case is trivial (we believe).

So ideally I think we'd want to only do this in the slow paths inside jl_task_get_next?

BUT to be honest, I find your approach much cleaner and clearer - and it's nice that we _would_ get an accurate measure of that switch time which is supposed to be cheap......... AND we are going to need something like this anyway for the per-task CPU time in https://relationalai.atlassian.net/browse/RAI-30482...

So maybe if we could solve the vDSO / fast instruction problem, then we _could_ do it this way after all?
Something to resolve CC @kpamnany.